flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/filesystem/FileSystemTableSinkTest.java

```diff
@@ -0,0 +1,214 @@
+            Assert.assertTrue(
+                    ExceptionUtils.findThrowableWithMessage(
+                                    e, "is configured, which is different from input parallelism")
+                            .isPresent());
+        }
+    }
+
+    @Test
+    public void testFileSystemTableSinkWithParallelismInStreaming() {
+
+        int parallelism = 2;
+
+        DescriptorProperties descriptor = new DescriptorProperties();
+        descriptor.putString(FactoryUtil.CONNECTOR.key(), "filesystem");
+        descriptor.putString("path", "/tmp");
+        descriptor.putString("format", "testcsv");
+        descriptor.putString(FactoryUtil.SINK_PARALLELISM.key(), String.valueOf(parallelism));
+
+        final DynamicTableSink tableSink = createSink(descriptor);
+        Assert.assertTrue(tableSink instanceof FileSystemTableSink);
+
+        final DynamicTableSink.SinkRuntimeProvider provider =
+                tableSink.getSinkRuntimeProvider(new MockSinkContext(false));
+        Assert.assertTrue(provider instanceof DataStreamSinkProvider);
+
+        final DataStreamSinkProvider dataStreamSinkProvider = (DataStreamSinkProvider) provider;
+        final DataStreamSink<?> dataStreamSink =
+                dataStreamSinkProvider.consumeDataStream(createInputDataStream());
+        final List<Transformation<?>> inputs = dataStreamSink.getTransformation().getInputs();
+        Assert.assertTrue(inputs.get(0).getParallelism() == parallelism);
+    }
+
+    @Test
+    public void testFileSystemTableSinkWithParallelismInBatch() {
+
+        int parallelism = 2;
+
+        DescriptorProperties descriptor = new DescriptorProperties();
+        descriptor.putString(FactoryUtil.CONNECTOR.key(), "filesystem");
+        descriptor.putString("path", "/tmp");
+        descriptor.putString("format", "testcsv");
+        descriptor.putString(FactoryUtil.SINK_PARALLELISM.key(), String.valueOf(parallelism));
+
+        final DynamicTableSink tableSink = createSink(descriptor);
+        Assert.assertTrue(tableSink instanceof FileSystemTableSink);
+
+        final DynamicTableSink.SinkRuntimeProvider provider =
+                tableSink.getSinkRuntimeProvider(new MockSinkContext(true));
+        Assert.assertTrue(provider instanceof DataStreamSinkProvider);
+
+        final DataStreamSinkProvider dataStreamSinkProvider = (DataStreamSinkProvider) provider;
+        final DataStreamSink<?> dataStreamSink =
+                dataStreamSinkProvider.consumeDataStream(createInputDataStream());
+        Assert.assertTrue(dataStreamSink.getTransformation().getParallelism() == parallelism);
+    }
+
+    private static DataStream<RowData> createInputDataStream() {
+        final MockTransformation<RowData> mockTransformation =
+                MockTransformation.createMockTransformation();
+        final DummyStreamExecutionEnvironment mockEnv = new DummyStreamExecutionEnvironment();
+
+        return new DataStream<>(mockEnv, mockTransformation);
```

Comment:
We should [Avoid Mock](https://flink.apache.org/contributing/code-style-and-quality-common.html) as much as possible, because it tends to be costly to maintain in the long run. 

Creating an input DataStream is very easy by `env.fromCollection(Collection<OUT> data, TypeInformation<OUT> typeInfo)`.